crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -700,7 +700,10 @@ impl<'a> Resolver<'a> {
             self.push_err(ResolverError::DistinctNotAllowed { ident: func.name_ident().clone() });
         }
 
-        if attributes == Some(Attribute::Test) && !parameters.is_empty() {
+        if ((attributes == Some(Attribute::Test { expect_failure: true }))
+            || (attributes == Some(Attribute::Test { expect_failure: false })))
+            && !parameters.is_empty()
```

Comment:
```suggestion
        if matches!(attributes, Some(Attribute::Test { .. })) && !parameters.is_empty()
```
Using `matches` should clean this up